Source/WebCore/platform/graphics/gstreamer/GLVideoSinkGStreamer.cpp

```diff
@@ -57,6 +63,88 @@ static GstStaticPadTemplate sinkTemplate = GST_STATIC_PAD_TEMPLATE("sink", GST_P
 WEBKIT_DEFINE_TYPE_WITH_CODE(WebKitGLVideoSink, webkit_gl_video_sink, GST_TYPE_BIN,
     GST_DEBUG_CATEGORY_INIT(webkit_gl_video_sink_debug, "webkitglvideosink", 0, "GL video sink element"))
 
+#if USE(GBM)
+static bool s_isDMABufDisabled;
+
+static void initializeDMABufAvailability()
+{
+    static std::once_flag onceFlag;
+    std::call_once(onceFlag, [] {
+        if (!webkitGstCheckVersion(1, 20, 0))
+            return;
+
+        const char* value = g_getenv("WEBKIT_GST_DMABUF_SINK_DISABLED");
+        auto valueSpan = WTF::span(value);
+        s_isDMABufDisabled = value && (equalLettersIgnoringASCIICase(valueSpan, "true"_s) || equalLettersIgnoringASCIICase(valueSpan, "1"_s));
+        if (!s_isDMABufDisabled && !DRMDeviceManager::singleton().mainGBMDeviceNode(DRMDeviceManager::NodeType::Render))
+            s_isDMABufDisabled = true;
+    });
+}
+
+static GRefPtr<GstCaps> buildDMABufCaps()
+{
+    GRefPtr<GstCaps> caps = adoptGRef(gst_caps_from_string("video/x-raw(memory:DMABuf), width = " GST_VIDEO_SIZE_RANGE ", height = " GST_VIDEO_SIZE_RANGE ", framerate = " GST_VIDEO_FPS_RANGE));
+#if GST_CHECK_VERSION(1, 24, 0)
+    gst_caps_set_simple(caps.get(), "format", G_TYPE_STRING, "DMA_DRM", nullptr);
+
+    static const char* formats = getenv("WEBKIT_GST_DMABUF_FORMATS");
+    if (formats && *formats) {
+        GUniquePtr<char*> tokens(g_strsplit(formats, ",", -1));
```

Comment:
You could use StringView and plit.

Reply:
I wanted to avoid utf8 -> utf16 -> utf8 conversions.